Userland/Libraries/LibWeb/HTML/NavigableContainer.cpp

```diff
@@ -262,25 +262,27 @@ void NavigableContainer::destroy_the_child_navigable()
     // 3. Set container's content navigable to null.
     m_content_navigable = nullptr;
 
-    // 4. Destroy navigable's active document.
+    // FIXME: 4. Inform the navigation API about child navigable destruction given navigable.
+
+    // 5. Destroy navigable's active document.
     navigable->active_document()->destroy();
 
-    // 5. Let parentDocState be container's node navigable's active session history entry's document state.
-    auto parent_doc_state = this->navigable()->active_session_history_entry()->document_state;
+    // 6. Let parentDocState be container's node navigable's active session history entry's document state.
+    auto parent_doc_state = navigable->active_session_history_entry()->document_state;
 
-    // 6. Remove the nested history from parentDocState's nested histories whose id equals navigable's id.
+    // 7. Remove the nested history from parentDocState's nested histories whose id equals navigable's id.
     parent_doc_state->nested_histories().remove_all_matching([&](auto& nested_history) {
         return navigable->id() == nested_history.id;
     });
 
-    // 7. Let traversable be container's node navigable's traversable navigable.
-    auto traversable = this->navigable()->traversable_navigable();
+    // 8. Let traversable be container's node navigable's traversable navigable.
+    auto traversable = navigable->traversable_navigable();
```

Comment:
Here `this->navigable()` is intentional because `navigable` variable is corresponding to "content navigable" while it should be "container's node navigable" like spec comment says.

Reply:
Whoops! Restored that one.